api/src/services/users.js

```diff
@@ -625,6 +651,12 @@ module.exports = {
               settings.contact_id = null;
             }
           })
+          .then(() => isDbAdmin(user))
```

Comment:
No specific reasons, I thought about it but didn't find much different on putting it before or after. 
Having the password in the user doc doesn't conflict when login, but since couch isn't adding the admin password in the docs (not sure if because of security) I thought on keeping it the same behaviour. 
I'm going to move this "is admin?" check to inside the update admin function.

Reply:
If we do it `after`, we make sure we don't update the password if the `PUT` fails, making the change more atomic. 
If we do it `before`, we could end up with a successful partial update of the password and the failure for the rest of the fields if the `PUT` fails.

Reply:
Done, so I have refactored a bit this function to make it flatter and easier to read, I hope it's okay